scripts/core/SuperdeskReactComponent.tsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import {httpRequestJsonLocal} from 'core/helpers/network';
+import {noop} from 'lodash';
+
+export class SuperdeskReactComponent<IProps = {}, IState = {}> extends React.PureComponent<IProps, IState> {
+    private abortControllers: Array<AbortController>;
+
+    /**
+     * Will automatically abort in-progress asynchronous operations(only those in asyncHelpers) when unmounting.
+     */
+    public asyncHelpers: {
+        fetch: (input: RequestInfo, init?: RequestInit) => Promise<Response>;
+        httpRequestJsonLocal: <T>(options: Parameters<typeof httpRequestJsonLocal>[0]) => Promise<T>;
+    };
+
+    constructor(props: IProps) {
+        super(props);
+
+        this.abortControllers = [];
+
+        this.asyncHelpers = {
+            fetch: (input: RequestInfo, init?: RequestInit): Promise<Response> => {
+                const controller = new AbortController();
+
+                this.abortControllers.push(controller);
+
+                // wrapping in another promise in order to handle AbortError in this component.
+                return new Promise((resolve, reject) => {
```

Comment:
wouldn't this work the same if you do `return fetch(...)` ?

Reply:
Nope. If a rejected promise is not returned in the catch block, the next `.then` gets executed.

```
new Promise(() => {
    throw new Error('aaa');
})
    .then(() => {
        console.log('log1');
    })
    .catch((err) => {
        console.log('catch', err.message);
    })
    .then(() => {
        console.log('log2');
    })
```

```
> catch aaa
> log2
```

If abort error occurs I want the promise chain to stop.

Reply:
isn't this how promises work? you can return `Promise.reject` in the catch to stop the chain if you want

Reply:
It is how they work, and I don't like it :smile: If I return `Promise.reject`, it will display an error in the console about an unhandled rejection and will likely send it to sentry too which shouldn't be done. Code that is behaving correctly should not produce errors.

I could add a `catch` handler when consuming the method like this:
```
componentDidMount() {
    if (this.props.selectedUserId != null) {
        this.asyncHelpers.httpRequestJsonLocal<IUser>({
            method: 'GET',
            path: `/users/${this.props.selectedUserId}`,
        }).then((selectedUser) => {
            this.setState({selectedUser});
        }).catch(() => {
            // < ----------------------------------------------------------------
        });
    }
}
```

But this goes against the encapsulation principle - abort error is an implementation detail of `asyncHelpers.httpRequestJsonLocal` and implementation details should not escape the abstraction layer. It would also add boilerplate which the component is created to avoid.

Reply:
np, makes sense. would be good to avoid that duplication:
```ts
const promiseWrapper = (promise) => new Promise((resolve, reject) => {
    promise.then(resolve)
        .catch((err) => ...);
};

return promiseWrapper(fetch(...));
return promiseWrapper(httpRequestJsonLocal...);
```